tests/class-two-factor-core.php

```diff
@@ -1581,4 +1581,56 @@ public function test_uninstall_removes_user_meta() {
 			'Provider was disabled due to uninstall'
 		);
 	}
+
+	public function test_can_disable_default_providers() {
+		$this->assertContains( 'Two_Factor_Email', array_keys( Two_Factor_Core::get_providers() ), 'Email provider is enabled by default' );
+
+		add_filter(
+			'two_factor_providers',
+			function( $providers ) {
+				return array_diff_key( $providers, array( 'Two_Factor_Email' => null ) );
+			}
+		);
+
+		$this->assertNotContains( 'Two_Factor_Email', array_keys( Two_Factor_Core::get_providers() ), 'Default provider can be disabled via a filter' );
```

Comment:
@ocean90 @r-a-y Here is a unit test to confirm that a core provider can be disabled via the filter.